src/main/java/net/minestom/server/extras/selfmodification/MinestomRootClassLoader.java

```diff
@@ -279,4 +311,23 @@ public void addURL(URL url) {
     public List<CodeModifier> getModifiers() {
         return modifiers;
     }
+
+    /**
+     * Tries to know which extension created this object, based on the classloader of the object. This can only check that the class of the object has been loaded
+     * by an extension.
+     *
+     * While not perfect, this should detect any callback created via extension code.
+     * It is possible this current version of the implementation might struggle with callbacks created through external
+     * libraries, but as libraries are loaded separately for each extension, this *should not*(tm) be a problem.
+     *
+     * @param obj the object to get the extension of
+     * @return <code>Optional.empty()</code> if no extension has been found, <code>Optional.of(&lt;name&gt;)</code> with 'name' being the extension name
+     */
+    public static Optional<String> findExtensionObjectOwner(@NotNull Object obj) {
```

Comment:
This should return a nullable string rather than an optional to fit style.

Reply:
Fixed in commit ee158c0deafeb6ded77496b32f5fde9f07f2f2ac